src/firestarter/Environment/X86/Payload/AVX512_AMX_Payload.cpp

```diff
@@ -0,0 +1,592 @@
+                temp_reg);
+        mov_dst++;
+      } else if (item == "L1_L") {
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmm2);
+        cb.vfmadd231pd(Zmm(add_dest), zmm1, zmmword_ptr(l1_addr, 64));
+        L1_INCREMENT();
+      } else if (item == "L1_BROADCAST") {
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmm2);
+        cb.vbroadcastsd(Zmm(add_dest), ptr_64(l1_addr, 64));
+        L1_INCREMENT();
+      } else if (item == "L1_S") {
+        cb.vmovapd(zmmword_ptr(l1_addr, 64), Zmm(add_dest));
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmm2);
+        L1_INCREMENT();
+      } else if (item == "L1_LS") {
+        cb.vmovapd(zmmword_ptr(l1_addr, 64), Zmm(add_dest));
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmmword_ptr(l1_addr, 128));
+        L1_INCREMENT();
+      } else if (item == "L2_L") {
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmm2);
+        cb.vfmadd231pd(Zmm(add_dest), zmm1, zmmword_ptr(l2_addr, 64));
+        L2_INCREMENT();
+      } else if (item == "L2_S") {
+        cb.vmovapd(zmmword_ptr(l2_addr, 64), Zmm(add_dest));
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmm2);
+        L2_INCREMENT();
+      } else if (item == "L2_LS") {
+        cb.vmovapd(zmmword_ptr(l2_addr, 64), Zmm(add_dest));
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmmword_ptr(l2_addr, 128));
+        L2_INCREMENT();
+      } else if (item == "L3_L") {
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmm2);
+        cb.vfmadd231pd(Zmm(add_dest), zmm1, zmmword_ptr(l3_addr, 64));
+        L3_INCREMENT();
+      } else if (item == "L3_S") {
+        cb.vmovapd(zmmword_ptr(l3_addr, 64), Zmm(add_dest));
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmm2);
+        L3_INCREMENT();
+      } else if (item == "L3_LS") {
+        cb.vmovapd(zmmword_ptr(l3_addr, 64), Zmm(add_dest));
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmmword_ptr(l3_addr, 128));
+        L3_INCREMENT();
+      } else if (item == "L3_P") {
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmmword_ptr(l1_addr, 64));
+        cb.prefetcht2(ptr(l3_addr));
+        L3_INCREMENT();
+      } else if (item == "RAM_L") {
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmm2);
+        cb.vfmadd231pd(ram_reg, zmm1, zmmword_ptr(ram_addr, 64));
+        RAM_INCREMENT();
+      } else if (item == "RAM_S") {
+        cb.vmovapd(zmmword_ptr(ram_addr, 64), Zmm(add_dest));
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmm2);
+        RAM_INCREMENT();
+      } else if (item == "RAM_LS") {
+        cb.vmovapd(zmmword_ptr(ram_addr, 64), Zmm(add_dest));
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmmword_ptr(ram_addr, 128));
+        RAM_INCREMENT();
+      } else if (item == "RAM_P") {
+        cb.vfmadd231pd(Zmm(add_dest), zmm0, zmmword_ptr(l1_addr, 64));
+        cb.prefetcht2(ptr(ram_addr));
+        RAM_INCREMENT();
+      } else if (item == "AMX") {
+      	    cb.tdpbf16ps(Tmm(counter%6), tmm6, tmm7);
+            counter++;
+      } else {
```

Comment:
Since this seems to be the only addition to the stock AVX512 payload can we integrate it there instead?

Reply:
I integrated it with the AVX512 payload. The check for the AMX_BF16 feature is done in compilePayload() function whenever AMX instructions are added to the mix.